lennonFolder/lennon-github-serach/simple-github/src/main/kotlin/com/androidhuman/example/simplegithub/ui/search/SearchAdapter.kt

```diff
@@ -0,0 +1,82 @@
+package com.androidhuman.example.simplegithub.ui.search
+
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.support.v7.widget.RecyclerView
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import com.androidhuman.example.simplegithub.R
+import com.androidhuman.example.simplegithub.api.model.GithubRepo
+import com.androidhuman.example.simplegithub.ui.GlideApp
+import java.util.*
+
+/**
+ * p.245 코틀린 익스텐션은 구글이 버리는것 같아서 적용하지 않았음. 대신 뷰 바인딩을 씁시다.
+ */
+class SearchAdapter : RecyclerView.Adapter<SearchAdapter.RepositoryHolder>() {
+    private var items: MutableList<GithubRepo> = mutableListOf()
+    private val placeholder = ColorDrawable(Color.GRAY)
+    private var listener: ItemClickListener? = null
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepositoryHolder =
+            RepositoryHolder(parent)
+
+
+    override fun onBindViewHolder(holder: RepositoryHolder, position: Int) {
+        items[position].let { repo ->
+            with(holder.itemView) {
+                GlideApp.with(context)
+                        .load(repo.owner.avatarUrl)
+                        .placeholder(placeholder)
+                        .into(holder.ivProfile)
+
+                holder.tvName.text = repo.fullName
+                holder.tvLanguage.text = if (TextUtils.isEmpty(repo.language))
+                    context.getText(R.string.no_language_specified)
+                else
+                    repo.language
+
+                setOnClickListener {
+                    if (null != listener) {
+                        listener!!.onItemClick(repo)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun getItemCount(): Int =
+            items.size
+
+    fun setItems(items: List<GithubRepo>) {
+        // 인자로 받은 리스트의 형태를 어댑터 내부에서 사용하는 리스트 형태(내부자료 변경이 가능한 형태)로 변환해주어야 한단당
+        this.items = items.toMutableList()
+    }
+
+    fun setItemClickListener(listener: ItemClickListener?) {
+        this.listener = listener
+    }
+
+    fun clearItems() {
+        items.clear()
+    }
+
+    class RepositoryHolder(parent: ViewGroup) : RecyclerView.ViewHolder(LayoutInflater.from(parent.context)
```

Comment:
onCreateView에서 view를 넘겨주면 아래처럼 수정 가능합니다~

```suggestion
class RepositoryHolder(view: View) : RecyclerView.ViewHolder(view)
```

Reply:
onCreateView에서 binding를 넘겨주면 아래처럼 수정 가능합니다~

```suggestion
class RepositoryHolder(binding: ItemRepositoryBinding) : RecyclerView.ViewHolder(binding.root)
```

Reply:
뷰바인딩으로 사용해보겠습니다.